src/Model/Behavior/SearchBehavior.php

```diff
@@ -53,8 +54,12 @@ public function findSearch(Query $query, array $options)
                 'to be nested under key "search" in find() options.'
             );
         }
-
-        $filters = $this->_getAllFilters();
+        $collection = Hash::get($options, 'collection');
+        if (is_string($collection) && !empty($collection)) {
+            $filters = $this->getFilters($collection);
```

Comment:
A new method isn't necessary. You can just add the collection argument to `_getFilters()` method and use a one liner `$filters = $this->_getAllFilters(Hash::get($options, 'collection'));`